physx/CMakeLists.txt

```diff
@@ -0,0 +1,185 @@
+##  * Neither the name of NVIDIA CORPORATION nor the names of its
+##    contributors may be used to endorse or promote products derived
+##    from this software without specific prior written permission.
+##
+## THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+## EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+## IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+## PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+## CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+## EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+## PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+## PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+## OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+## (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+## OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+##
+## Copyright (c) 2018-2019 NVIDIA Corporation. All rights reserved.
+## Copyright (c) 2019 Pablo Hernandez-Cerdan
+
+cmake_minimum_required(VERSION 3.13)
+# Change options for CMAKE_BUILD_TYPE and set default.
+# Same values than CMAKE_CONFIGURATION_TYPES
+# From https://gitlab.kitware.com/cmake/cmake/issues/19401
+get_property(multiConfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+if(NOT multiConfig AND NOT DEFINED CMAKE_BUILD_TYPE)
+    set(default_build_type "release")
+    message(STATUS "Setting build type to '${default_build_type}' as none was specified.")
+    set(CMAKE_BUILD_TYPE ${default_build_type} CACHE STRING "Choose the type of build.")
+    set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+        "release" "profile" "checked" "debug")
+endif()
+
+file(READ "version.txt" PHYSX_VERSION)
+project(PhysX
+    LANGUAGES C CXX
+    VERSION ${PHYSX_VERSION})
+message(STATUS "PhysX VERSION: ${PHYSX_VERSION}")
+cmake_policy(SET CMP0057 NEW) # Enable IN_LIST
+cmake_policy(SET CMP0077 NEW) # option() does nothing when variable is alredy set
+
+# Set a default build type if none was specified
+set(default_build_type "Release")
+if(EXISTS "${CMAKE_SOURCE_DIR}/.git")
+  set(default_build_type "Debug")
+endif()
+
+# PhysXSDK options:
+option(PX_BUILDSNIPPETS "Generate the snippets" OFF)
+option(PX_BUILDPUBLICSAMPLES "Generate the samples" OFF)
+option(PX_CMAKE_SUPPRESS_REGENERATION "Disable zero_check projects" OFF)
+# PhysX options:
+option(PX_SCALAR_MATH "Disable SIMD math" OFF)
+option(PX_GENERATE_STATIC_LIBRARIES "Generate static libraries" OFF)
+if(NOT DEFINED BUILD_SHARED_LIBS)
+    if(PX_GENERATE_STATIC_LIBRARIES)
+        set(BUILD_SHARED_LIBS OFF)
+    else()
+        set(BUILD_SHARED_LIBS ON)
+    endif()
+else()
+    if(BUILD_SHARED_LIBS EQUAL PX_GENERATE_STATIC_LIBRARIES)
+        message(FATAL_ERROR "Contradictory options: BUILD_SHARED_LIBS and PX_GENERATE_STATIC_LIBRARIES have both the same value: ${BUILD_SHARED_LIBS}")
+    endif()
+endif()
```

Comment:
What about adding this variable in CACHE so that downstream users can easily switch them with `ccmake` or cmake gui?

Reply:
Users can do that with the `PX_GENERATE_STATIC_LIBRARIES` option (stored in the CACHE already). I would prefer to keep changes from current NVidia setup at a minimum to facilitate future integration.